src/screens/NativeIntegration/components/NIInputField.tsx

```diff
@@ -0,0 +1,212 @@
+import type { Collectible } from 'models/Collectible';
+
+// Constants
+import { ASSET_TYPES } from 'constants/assetsConstants';
+
+type Props = {
+  itemInfo: infoType;
+  valueInputRef?: React.Ref<typeof RNTextInput>;
+  value: any;
+  onChangeValue: (val: any) => void;
+  txFeeInfo?: any;
+  chain?: string;
+  disableAssetSelectorModal?: boolean;
+};
+
+type infoType = {
+  internalType: string;
+  name: string | null | '';
+  type: string;
+};
+
+function NIInputField({ itemInfo, chain, value, onChangeValue, txFeeInfo, disableAssetSelectorModal = false }: Props) {
+  const { t } = useTranslation();
+  const inputRef = React.useRef();
+  const colors = useThemeColors();
+  const activeAccountAddress = useRootSelector(activeAccountAddressSelector);
+  const assetsWithBalance = useRootSelector(accountAssetsWithBalanceSelector);
+
+  let defaultAssetData = assetsWithBalance.find(
+    (response) => response?.chain === chain && response?.address === '0x0000000000000000000000000000000000000000',
+  );
+
+  const defaultAssetOption = defaultAssetData && {
+    ...defaultAssetData,
+    symbol: defaultAssetData.token,
+  };
+
+  const [assetData, setAssetData] = React.useState<AssetOption | Collectible>(
+    defaultAssetData?.tokenType ? defaultAssetData : defaultAssetOption || [],
+  );
+  const selectedToken: AssetOption = assetData?.tokenType !== ASSET_TYPES.COLLECTIBLE ? assetData : null;
+  const selectedCollectible: Collectible = assetData?.tokenType === ASSET_TYPES.COLLECTIBLE ? assetData : null;
+
+  const handleSelectToken = (token: AssetOption) => {
+    if (token) setAssetData(token);
+  };
+
+  const handleSelectCollectible = (collectible: Collectible) => {
+    if (collectible) setAssetData?.(collectible);
+  };
+
+  const getValidationError = () => {
+    if (value?.c || typeof value == 'boolean') return null;
+    if (addressesEqual(value, activeAccountAddress)) {
+      return t('error.cannotSendToYourself');
+    }
+    if (value && !isValidAddressOrEnsName(value)) {
+      return t('error.incorrectAddress');
+    }
+    return null;
+  };
+
+  const onChangeAddress = (val) => {
+    const input = val.trimLeft().replace(/\s\s/g, ' ');
+    onChangeValue(input);
+  };
+
+  const errorMessage = getValidationError();
+  const inputStyles = [styles.input, { color: !!errorMessage ? colors.negative : colors.text }];
+
+  const inputComponent = () => {
+    if (itemInfo?.type === 'nativeTokenInput') {
+      return (
+        <AssetSelector
+          selectedToken={selectedToken}
+          onSelectToken={handleSelectToken}
+          selectedCollectible={selectedCollectible}
+          onSelectCollectible={handleSelectCollectible}
+          value={value}
+          onValueChange={onChangeValue}
+          txFeeInfo={txFeeInfo}
+          disableAssetSelectorModal={disableAssetSelectorModal}
+          hideFiatValueInput
+        />
+      );
+    }
+
+    if (itemInfo?.type === 'uint256') {
```

Comment:
There are several variations of `uint` here, with 256 being one of them. This data type starts from `uint8` to `uint256`, do you think a regex is better here?

Reply:
Okay, I'll updated it.